subworkflows/local/preprocessing.nf

```diff
@@ -0,0 +1,83 @@
+nextflow.enable.dsl = 2
+
+//includes
+include { FORCE_GENERATE_TILE_ALLOW_LIST }         from '../../modules/local/force-generate_tile_allow_list'
+include { FORCE_GENERATE_ANALYSIS_MASK }           from '../../modules/local/force-generate_analysis_mask'
+include { PREPROCESS_CONFIG }                      from '../../modules/local/preprocess_force_config'
+include { FORCE_PREPROCESS }                       from '../../modules/local/force-preprocess'
+include { MERGE as MERGE_BOA; MERGE as MERGE_QAI } from '../../modules/local/merge'
+
+//Closure to extract the parent directory of a file
+def extractDirectory = { it.parent.toString().substring(it.parent.toString().lastIndexOf('/') + 1 ) }
+
+workflow PREPROCESSING {
+
+    take:
+        data
+        dem
+        wvdb
+        cube_file
+        aoi_file
+
+    main:
+
+        ch_versions = Channel.empty()
+
+        FORCE_GENERATE_TILE_ALLOW_LIST( aoi_file, cube_file )
+        ch_versions = ch_versions.mix(FORCE_GENERATE_TILE_ALLOW_LIST.out.versions)
+
+        FORCE_GENERATE_ANALYSIS_MASK( aoi_file, cube_file )
+        ch_versions = ch_versions.mix(FORCE_GENERATE_ANALYSIS_MASK.out.versions)
+
+        //Group masks by tile
+        masks = FORCE_GENERATE_ANALYSIS_MASK.out.masks.flatten().map{ x -> [ extractDirectory(x), x ] }
+
+        // Preprocessing configuration
+        PREPROCESS_CONFIG( data, cube_file, FORCE_GENERATE_TILE_ALLOW_LIST.out.tile_allow, dem, wvdb )
+        ch_versions = ch_versions.mix(PREPROCESS_CONFIG.out.versions.first())
+
+        // Main preprocessing
+        FORCE_PREPROCESS( PREPROCESS_CONFIG.out.preprocess_config_and_data)
+        ch_versions = ch_versions.mix(FORCE_PREPROCESS.out.versions.first())
+
+        //Group by tile, date and sensor
+        boa_tiles = FORCE_PREPROCESS.out.boa_tiles.flatten().map{ [ "${extractDirectory(it)}_${it.simpleName}", it ] }.groupTuple()
```

Comment:
In n-core we usually don't encode information as directory/file names, but instead use a [meta map](https://nf-co.re/docs/contributing/components/meta_map)

Reply:
I'm aware of that. The reasons we decided to maintain the name-encoded information is that it is the common approach in remote sensing and somewhat expected by FORCE. I will look into switching to meta maps.

Reply:
I think it's fine to encode it in file names if it's common/the standard in the field - I would say it's not a blocker for this release.

But if that's the case, I think it would be important to add validation checks to ensure that the file name structure _is_ exactly as expected for the pipeline.

But of course it wouldn't hurt to copy information like that into a meta.map to accompany the files through the pipeline.